packages/dashboard/src/components/appbar.tsx

```diff
@@ -51,6 +60,12 @@ export default function AppBar(props: AppBarProps): React.ReactElement {
         </IconButton>
         {user && (
           <>
+            <EmergencyAlarm
+              isActive={alarmState}
+              onTurnOn={() => emergencyAlarmRequestPub?.publish({ data: true })}
```

Comment:
To make this more testable and reduce coupling, use a callback instead of adding a dependency on `transport`.

Reply:
Sounds good, thanks! https://github.com/osrf/romi-dashboard/pull/129/commits/771b4b4755692e3b0cad9b8ff1395f20a5afe7fa